### PR TITLE
Toast action can be a link

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- toast `action` accepts `ILinkProps` (href and target), uses `AnchorButton`
- refactor toast tests to use a little wrapper function that find the action and dismiss buttons. this puts the queries in one place, if we need to change them in the future.
- ban `describe.only` and `it.only` in TSLint for the same reason as console.log: useful in local development but problematic in real environments.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -1,7 +1,11 @@
 {
   "extends": ["tslint:latest", "tslint-react"],
   "rules": {
-    "ban": [true, ["Object", "assign", "use TS2.1 object spread { ...a, ...b }"]],
+    "ban": [true,
+      ["Object", "assign", "use TS2.1 object spread { ...a, ...b }"],
+      ["describe", "only"],
```
Comment (adidahiya) [inline tslint.json]: nice 👍

File: packages/core/test/toast/toastTests.tsx [hunk 2]
```diff
@@ -9,45 +9,55 @@ import { assert } from "chai";
...
+        wrap(<Toast action={{ text: "Undo" }} message="Hello" onDismiss={handleDismiss} />)
+            .action.simulate("click");
         assert.isTrue(handleDismiss.calledOnce, "onDismiss not called once");
         assert.isTrue(handleDismiss.calledWith(false), "onDismiss not called with false");
     });
 
+    function wrap(toast: JSX.Element) {
+        const root = shallow(toast);
+        return {
+            action: root.find(AnchorButton),
```
Comment (adidahiya) [inline packages/core/test/toast/toastTests.tsx]: I feel like we should be querying for components by class rather than tagname, hm?

Comment (giladgray) [inline packages/core/test/toast/toastTests.tsx]: class isn't as easy as you'd hope. with shallow mounting you need the exact string, or a custom predicate. with full mounting, querying for `".pt-button"` gives you the HTML element, not the React component, so you can't do React-y things to it. 😢

File: packages/core/examples/toastExample.tsx [hunk 3]
```diff
@@ -32,11 +32,13 @@ export class ToastExample extends BaseExample<IToasterProps> {
 
     private TOAST_BUILDERS: IToastDemo[] = [{
         action: {
+            href: "https://www.google.com/search?q=toast&source=lnms&tbm=isch",
+            target: "_blank",
```
Comment (tgreenwatts) [inline packages/core/examples/toastExample.tsx]: Is this optional or will it always open in a new tab? Prefer to give users the option but not be opinionated

Comment (giladgray) [inline packages/core/examples/toastExample.tsx]: optional. this is just an example usage.

**Pull request merged**